processor/elasticinframetricsprocessor/processor.go

```diff
@@ -60,13 +62,49 @@ func (p *ElasticinframetricsProcessor) processMetrics(_ context.Context, md pmet
 	for i := 0; i < md.ResourceMetrics().Len(); i++ {
 		resourceMetric := md.ResourceMetrics().At(i)
 		rm := resourceMetric.Resource()
-
 		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
 			scopeMetric := resourceMetric.ScopeMetrics().At(j)
 			for _, r := range p.remappers {
 				r.Remap(scopeMetric, scopeMetric.Metrics(), rm)
 			}
 		}
 	}
+	// override=True will keep only the metrics that have been remapped based on the presense of OTelRemappedLabel label.
+	// See  https://github.com/elastic/opentelemetry-lib/blob/6d89cbad4221429570107eb4a4968cf8a2ff919f/remappers/common/const.go#L31
+	if p.cfg.Override {
+		newmetic := pmetric.NewMetrics()
+		rmnew := newmetic.ResourceMetrics().AppendEmpty()
+		rmscope := rmnew.ScopeMetrics().AppendEmpty()
+		for i := 0; i < md.ResourceMetrics().Len(); i++ {
```

Comment:
Although, it's a matter of choice, using `range` makes it look a little cleaner, while still be able to use the index. 
```suggestion
		for i := range md.ResourceMetrics() {
```
Also, another small suggestion; instead of i, j, l, and m, we could use the names that reflect what these indices represent to enhance readability e.g. 
```suggestion
		for resIdx := range md.ResourceMetrics() {
```

Reply:
Done ! thanks you !!! let me know if is ok now

Reply:
Thanks for making the change. I think we can simply range over the returned value of `md.ResourceMetrics()`. Currently, it ranges over len. WDYT?

Reply:
```bash
cannot range over md.ResourceMetrics() (value of type pmetric.ResourceMetricsSlice)
```

So not possible :)